pkg/security/ebpf/kprobe_bpf.go

```diff
@@ -13,26 +13,32 @@ import (
 	"strings"
 	"syscall"
 	"time"
+
+	"github.com/DataDog/datadog-agent/pkg/util/log"
+	"github.com/avast/retry-go"
 )
 
 const (
 	// maxEnableRetry number of retry for resource busy fail
 	maxEnableRetry = 3
 )
 
-func (m *Module) tryEnableKprobe(secName string) (err error) {
-	for i := 0; i != maxEnableRetry; i++ {
-		if err = m.EnableKprobe(secName, 512); err == nil {
-			break
+func (m *Module) tryEnableKprobe(secName string) error {
+	return retry.Do(func() error {
+		err := m.EnableKprobe(secName, 512)
+		if err == nil {
+			return nil
 		}
+
 		// not available, not a temporary error
-		if strings.Contains(err.Error(), syscall.ENOENT.Error()) {
-			break
+		if strings.Contains(err.Error(), syscall.ENOENT.Error()) ||
+			strings.Contains(err.Error(), syscall.EINVAL.Error()) {
+			return nil
 		}
```

Comment:
Why do we retry enabling kprobes? And is this something that should be added to datadog/ebpf? 


Also: EINVAL is a pretty vague error, so this seems like something worth logging at the WARN level.

Reply:
EINVAL is returned when the sycall does not exist which is not an error for the runtime security agent